src/avram/criteria.cr

```diff
@@ -25,12 +25,35 @@ class Avram::Criteria(T, V)
   private def check_nilable!(_value : Nil)
     {% raise <<-ERROR
 
-      The 'eq' method cannot compare values that may be 'nil'
+      The 'eq' method cannot compare values that might be 'nil'.
 
-      Try this...
+      ▸ If you're ok comparing this column to 'nil'...
 
-        ▸ Use 'nilable_eq'. This method allows checking against nilable values
-        ▸ Ensure the value is not nilable by wrapping it in an 'if' or using 'not_nil!'
+          # Use 'nilable_eq' to allow querying against nilable/NULL values.
+          #
+          # For example if you have a 'name' column and you want to allow
+          # people to query with a 'String' to find people with a given name, or 'nil'
+          # to find people without a name:
```

Comment:
This feels oddly phrased to me. Also, people without a name might be a strange thing. How about a `User`'s `invited_by_id`?

---

> For example if you have a User model with an 'invited_by_id' column, to find all people invited by a specific user:
UserQuery.new.invited_by_id.nilable_eq(42)

Reply:
Yeah you are right. Since name would almost always be not nil this does seem odd. I also think the invited_by_id has a similar problem. How about nickname? Since that is something that seems like you might want to check for a specific nickname or people with no nickname?

Reply:
Yeah, `nickname` works!

Reply:
👍